src/workerd/jsg/modules.c++

```diff
@@ -82,7 +82,11 @@ v8::MaybeLocal<v8::Module> resolveCallback(v8::Local<v8::Context> context,
       // as parent. We have to do it this way just in case the worker bundle
       // is using the prefix itself. (which isn't likely but is possible).
       // We only need to do this if internalOnly is false.
-      if (!internalOnly && (spec.startsWith("node:") || spec.startsWith("cloudflare:"))) {
+      // The __STATIC_CONTENT_MANIFEST module is generated by Worker Sites.
+      if (!internalOnly && (spec.startsWith("node:") ||
+                            spec.startsWith("cloudflare:") ||
+                            spec == "__STATIC_CONTENT_MANIFEST")
```

Comment:
non-blocking nit: might be worthwhile separating this second set of conditions into a separate utility function for better readability/managability.

e.g. 

```cpp
bool isSpecialModule(kj::StringPtr spec) {
  return spec.startsWith("node:") ||
             spec.startsWith("cloudflare:") ||
             spec == "__STATIC_CONTENT_MANIFEST";
}

if (!internalOnly && isSpecialModule(spec)) {
  // ...
}
```

That said, there is an ever so unlikely chance that this could break someone. Specifically, while it is exceedingly unlikely, an existing worker could theoretically have an existing module named `__STATIC_CONTENT_MANIFEST`. Have you verified yet that this won't break folks? If not, we may first want to roll out a temporary log statement that rules it out.

Reply:
noted, will refactor once the PR get consensus.